### PR TITLE
Fix sparse mul!

### LINKED ISSUE #629: mul! failing for adjoint sparse matrices
MWE 
```Julia
using CuArrays
using SparseArrays
using LinearAlgebra

const n = 100

M = CuArray(diagm(0=>ones(n), 1=>-1*ones(n-1), -1=>-1*ones(n-1)))
M2 = sparse(M);
V = CuArray(rand(100));
adjoint(M2)*V
```
yields 

```
UndefVarError: transA not defined

Stacktrace:
 [1] mul!(::CuArray{Float64,1,Nothing}, ::Adjoint{Float64,CuArrays.CUSPARSE.CuSparseMatrixCSR{Float64}}, ::CuArray{Float64,1,Nothing}) at /home/USERNAME/.julia/packages/CuArrays/A6GUx/src/sparse/interfaces.jl:14
 [2] *(::Adjoint{Float64,CuArrays.CUSPARSE.CuSparseMatrixCSR{Float64}}, ::CuArray{Float64,1,Nothing}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/LinearAlgebra/src/matmul.jl:113
 [3] top-level scope at In[3]:1
```

just from inspecting the code, this seems to me like an obvious error in the code, or isn't it?

it is:
```Julia
LinearAlgebra.mul!(C::CuVector{T},adjA::Adjoint{<:Any,<:CuSparseMatrix},B::CuVector) where {T} = mv!('C',one(T),parent(transA),B,zero(T),C,'O')
```
but it should be 
```Julia
LinearAlgebra.mul!(C::CuVector{T},adjA::Adjoint{<:Any,<:CuSparseMatrix},B::CuVector) where {T} = mv!('C',one(T),parent(adjA),B,zero(T),C,'O')
``` 
or is there more to it? It did not try it out yet.

### PR DESCRIPTION
**Pull request by amontoison**
close #629 
#630 , #637 
@haampie

### CODE REVIEW

**Comment by maleadt** [comment]
Thanks!
bors r+

**Comment by amontoison** [comment]
Tests failed with `A' * x` when A is a `CuSparseMatrixCSC{ComplexF32}` or a `CuSparseMatrixCSC{ComplexF64}`. All tests passed with `CuSparseMatrixCSR{...}`.

**Comment by amontoison** [comment]
The culprit should be `mv!` : https://github.com/JuliaGPU/CuArrays.jl/blob/master/src/sparse/wrappers.jl#L554-L584

**Comment by amontoison** [comment]
The conjugate-transpose of a transposed matrix A is not A but `conj(A)`.

**Comment by amontoison** [comment]
I read the documentation of `csrmv` function and it seems that they forgot this corner case. 
https://docs.nvidia.com/cuda/cusparse/index.html#csrmv

@maleadt, it's ready to be tested.

File: src/sparse/wrappers.jl [hunk 1]
```diff
@@ -567,17 +567,22 @@ for (fname,elty) in ((:cusparseScsrmv, :Float32),
                 ctransa = 'T'
             end
             cutransa = cusparseop(ctransa)
-            cuind    = cusparseindex(index)
```
Comment (amontoison) [inline src/sparse/wrappers.jl:570]: `cuind` variable is not used. I removed it.

**Comment by maleadt** [comment]
bors d+

**Comment by maleadt** [comment]
bors try

**Comment by maleadt** [comment]
What about:

> [[DEPRECATED]] use cusparseSpMV() instead. The routine will be removed in the next major release

**Comment by maleadt** [comment]
bors try

**Comment by amontoison** [comment]
About the `DEPRECATED` csrmv, it concerned numerous functions in CUSPARSE module. :worried:

**Pull request merged**